### PR TITLE
(SERVER-2360) Update jruby-utils to 2.3.1

### PR DESCRIPTION
**Pull request by tvpartytonight**
This adds support for the multithreaded key for JRubyConfig.

### CODE REVIEW

**Comment by jonathannewman** [comment]
@tvpartytonight I think you meant "jruby-utils" not "clj-parent" in the commit title?

File: project.clj [hunk 1]
```diff
@@ -125,7 +125,7 @@
                          [puppetlabs/rbac-client ~rbac-client-version :classifier "test"]
                          [puppetlabs/analytics-client "1.0.2"]
                          [puppetlabs/clj-shell-utils "1.0.2"]
-                         [puppetlabs/jruby-utils "2.2.0"]
+                         [puppetlabs/jruby-utils "2.3.0"]
```
Comment (justinstoller) [inline project.clj]: Shouldn't this be 2.3.1? It looks like 2.3.0 is one that I updated (but forgot to actually release) that includes JRuby 9.2.9? https://github.com/puppetlabs/jruby-utils/commits/master

**Comment by tvpartytonight** [comment]
@jonathannewman @justinstoller updated.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## [4.2.7]
+
+- update jruby-utils to 2.3.0 to support usage of the multithreaded key for JRubyConfig
```
Comment (justinstoller) [inline CHANGELOG.md]: This is wrong now.

**Pull request merged**